src/index.cpp

```diff
@@ -7486,29 +7492,133 @@ void Index::batch_embed_fields(std::vector<index_record*>& records,
+                            std::vector<float> embedding_vals;
+                            if(value_to_embed.first->is_update) {
+                                if(value_to_embed.first->new_doc.count(field.name) == 0) {
+                                    embedding_vals = embeddings_text[i].embedding;
+                                } else {
+                                    std::vector<float> existing_embedding = value_to_embed.first->new_doc[field.name].get<std::vector<float>>();
+                                    // accumulate embeddings
+                                    for(size_t j = 0; j < existing_embedding.size(); j++) {
+                                        existing_embedding[j] += embeddings_text[i].embedding[j];
+                                    }
+                                    embedding_vals = existing_embedding;
+                                }
+                            } else {
+                                if(value_to_embed.first->doc.count(field.name) == 0) {
+                                    embedding_vals = embeddings_text[i].embedding;
+                                } else {
+                                    std::vector<float> existing_embedding = value_to_embed.first->doc[field.name].get<std::vector<float>>();
+                                    // accumulate embeddings
+                                    for(size_t j = 0; j < existing_embedding.size(); j++) {
+                                        existing_embedding[j] += embeddings_text[i].embedding[j];
+                                    }
+                                    embedding_vals = existing_embedding;
+                                }
+                            }
+
+                            if(value_to_embed.first->is_update) {
+                                value_to_embed.first->new_doc[field.name] = embedding_vals;
+                            } else {
+                                value_to_embed.first->doc[field.name] = embedding_vals;
+                            }
+                            count++;
+                        }
+                    }
+                }
+            }
+
+            if(!values_to_embed_image.empty()) {
+                for(size_t i = 0; i < values_to_embed_image.size(); i++) {
+                    auto& value_to_embed = values_to_embed_image[i];
+                    if(record == value_to_embed.first) {
+                        if(!value_to_embed.first->embedding_res.empty()) {
+                            continue;
+                        }
+                        if(!embeddings_image[i].success) {
+                            value_to_embed.first->embedding_res = embeddings_image[i].error;
+                            value_to_embed.first->index_failure(embeddings_image[i].status_code, "");
+                        } else {
+                            std::vector<float> embedding_vals;
+                            if(value_to_embed.first->is_update) {
```

Comment:
```
const auto& doc = value_to_embed.first->is_update ? value_to_embed.first->new_doc : value_to_embed.first->doc;
```

By extracting the doc out, the rest of the code for accumulation need not repeat.